### PR TITLE
bump luarocks packages

### PR DESCRIPTION
**Pull request by teto**
## Description of changes

WIP: bump luarocks package-set . Waiting to fix several luarocks upstream
before starting

- [x] lua-iconv: moved to lunarmodules with fixed rockspecs https://github.com/luarocks/luarocks-site/issues/199#issuecomment-1717484507
- [x] mimetypes: moved to lunarmodules with fixed rockspecs  
- [x] basexx: patch merged, we can use upstream again 
- [x] https://gitlab.com/peterbillam/pjb_lua/-/issues/2 (for now I've just removed it from the generated set and put directly in overrides)
- [x] https://github.com/JorjBauer/lua-cyrussasl/issues/6 wrong source, for now I've dropped it
- luarocks-nix: unstable-2023-02-26 -> unstable-2023-09-08
- [x] generateLuarocksConfig: add defaults to ease its usage
- wip

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/lua-modules/overrides.nix [hunk 1]
```diff
@@ -270,6 +270,13 @@ with prev;
   });
 
   lua-iconv = prev.lua-iconv.overrideAttrs (oa: {
+    # rockspec is broken https://github.com/ittner/lua-iconv/issues/8
+    src = fetchFromGitHub {
```
Comment (alerque) [inline pkgs/development/lua-modules/overrides.nix:274]: This workaround should be obsolete, you can go back to just using the default LuaRock.

**Comment by teto** [comment]
current status: 
```
32 packages marked as broken and skipped:
lua51Packages.digestif lua51Packages.lmpfrlib lua52Packages.digestif lua52Packages.gitsigns-nvim lua52Packages.lmpfrlib lua52Packages.telescope-manix lua52Packages.telescope-nvim lua53Packages.gitsigns-nvim lua53Packages.telescope-manix lua53Packages.telescope-nvim lua54Packages.cyan lua54Packages.gitsigns-nvim lua54Packages.haskell-tools-nvim lua54Packages.http lua54Packages.ldoc lua54Packages.luaevent lua54Packages.lush-nvim lua54Packages.nvim-cmp lua54Packages.plenary-nvim lua54Packages.readline lua54Packages.sqlite lua54Packages.teal-language-server lua54Packages.telescope-manix lua54Packages.telescope-nvim lua54Packages.tl luaPackages.digestif luaPackages.gitsigns-nvim luaPackages.lmpfrlib luaPackages.telescope-manix luaPackages.telescope-nvim luajitPackages.digestif luajitPackages.lmpfrlib

12 packages failed to build:
lua51Packages.lua-resty-openidc lua51Packages.std-normalize luaPackages.lua-resty-openidc luaPackages.std-normalize lua53Packages.lua-resty-openidc lua53Packages.std-normalize lua54Packages.lua-resty-openidc lua54Packages.std-normalize luajitPackages.lua-resty-openidc luajitPackages.std-normalize nelua openrussian-cli

1889 packages built:
```

File: pkgs/development/lua-modules/overrides.nix [hunk 2]
```diff
@@ -386,6 +387,11 @@ with prev;
     ];
   });
 
+  # lua-resty-session =  prev.lua-resty-session.overrideAttrs (oa: {
+  #   # lua_pack and lua-ffi-zlib are unpackaged, causing this package to not evaluate
+  #   meta.broken = true;
+  # });
```
[the remark this answers is not in the document]
Comment (mrcjkb) [reply pkgs/development/lua-modules/overrides.nix:393]: It looks like it's already marked broken on `master`?

**Comment by mrcjkb** [comment]
> I am tempted to merge this as is, as among the build failures, I am pretty sure most of them already fail on master, and it's simpler to fix the leaf packages with small rebuilds on master.

Took a closer look at the broken packages (on `x86_64-linux`) on `master`:

- `lua*Packages.lua-resty-openidc` is marked as broken.
- `lua*Packages.std-normalize` builds.
- `nelua` builds (only has `luacheck` as a lua dependency).
- `openrussian-cli` isn't marked as broken, but fails to build.

**Comment by teto** [comment]
thanks for the thoughtful review, as a follow up I've:
- marked openrussian-cli as broken
- updated nelua
- tried to fix std._normalize but the build failed even then,  we were using an unstable version that is broken so instead I told it to use the stable version and opened https://github.com/nix-community/luarocks-nix/pull/32 to prepare for the next release. I've packaged ldoc anyway
- lua*Packages.lua-resty-openidc can be adjusted easily once this reaches master for ZHF, but I wanted to have this PR in staging before 23.11 release.

File: pkgs/development/lua-modules/overrides.nix [hunk 3]
```diff
@@ -514,8 +520,19 @@ with prev;
...
+    version = "3.2-0";
+    knownRockspec = (fetchurl {
+      url    = "mirror://luarocks/readline-3.2-0.rockspec";
+      sha256 = "1r0sgisxm4xd1r6i053iibxh30j7j3rcj4wwkd8rzkj8nln20z24";
+    }).outPath;
+    src = fetchurl {
+      url    = "http://www.pjb.com.au/comp/lua/readline-3.2.tar.gz";
+      sha256 = "1mk9algpsvyqwhnq7jlw4cgmfzj30l7n2r6ak4qxgdxgc39f48k4";
+    };
+
```
[the remark this answers is not in the document]
Comment (teto) [reply pkgs/development/lua-modules/overrides.nix:528]: like for vimPlugins, the source is broken so I've stopped it from being generated. I've actually changed the rockspec url `url    = "http://www.pjb.com.au/comp/lua/readline-3.2.tar.gz";` to `url    = "http://pjb.com.au/comp/lua/readline-3.2.tar.gz";`

File: pkgs/development/lua-modules/overrides.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/lua-modules/overrides.nix]: indeed, I removed it

File: pkgs/development/lua-modules/overrides.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (teto) [reply pkgs/development/lua-modules/overrides.nix:386]: let's do this once this PR reaches master

**Pull request merged**